packages/twenty-chrome-extension/src/options/Sidepanel.tsx

```diff
@@ -41,126 +39,106 @@ const StyledActionContainer = styled.div`
 `;
 
 const Sidepanel = () => {
-  const [isAuthenticating, setIsAuthenticating] = useState(false);
   const [isAuthenticated, setIsAuthenticated] = useState(false);
-  const [iframeSrc, setIframeSrc] = useState(
+  const [clientUrl, setClientUrl] = useState(
     import.meta.env.VITE_FRONT_BASE_URL,
   );
-  const [error, setError] = useState('');
-  const [serverBaseUrl, setServerBaseUrl] = useState('');
-  const authenticate = () => {
-    setIsAuthenticating(true);
-    setError('');
-    chrome.runtime.sendMessage(
-      { action: 'launchOAuth' },
-      ({ status, message }) => {
-        if (status === true) {
-          setIsAuthenticated(true);
-          setIsAuthenticating(false);
-          chrome.storage.local.set({ isAuthenticated: true });
-        } else {
-          setError(message);
-          setIsAuthenticating(false);
-        }
-      },
-    );
-  };
+  const iframeRef = useRef<HTMLIFrameElement>(null);
+
+  const setIframeState = useCallback(async () => {
+    const store = await chrome.storage.local.get([
+      'isAuthenticated',
+      'sidepanelUrl',
+      'clientUrl',
+    ]);
+    if (store.isAuthenticated === true) setIsAuthenticated(true);
+
+    if (isDefined(store.sidepanelUrl)) {
+      isDefined(store.clientUrl)
+        ? setClientUrl(`${store.clientUrl}${store.sidepanelUrl}`)
+        : setClientUrl(`${clientUrl}${store.sidepanelUrl}`);
+    }
+  }, [clientUrl, setClientUrl]);
 
   useEffect(() => {
-    const getState = async () => {
-      const store = await chrome.storage.local.get();
-      if (isDefined(store.serverBaseUrl)) {
-        setServerBaseUrl(store.serverBaseUrl);
-      } else {
-        setServerBaseUrl(import.meta.env.VITE_SERVER_BASE_URL);
-      }
-
-      if (store.isAuthenticated === true) setIsAuthenticated(true);
-      const { tab: activeTab } = await chrome.runtime.sendMessage({
-        action: 'getActiveTab',
-      });
-
-      if (
-        isDefined(activeTab) &&
-        isDefined(store[`sidepanelUrl_${activeTab.id}`])
-      ) {
-        const url = store[`sidepanelUrl_${activeTab.id}`];
-        setIframeSrc(url);
-      }
-    };
-    void getState();
+    void setIframeState();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
We shouldn't disable this rule, it's a code smell, useEffect needs to have all its dependencies to behave correctly